src/odemis/driver/xt_client.py

```diff
@@ -2374,58 +2378,62 @@ def _moveTo(self, future: futures.Future, pos: Dict[str, float], rel: bool = Fal
+                    tol_rotation = max(min(movement_req_rotational), STAGE_TOL_ROTATION)  # radians
 
                 while check_pos:
                     current_pos = self.parent.get_stage_position()
                     # Every axis requested to move should have moved (compared to the position before starting)
                     # if there is some change w.r.t starting position, proceed to check the target position
-
-                    # TODO Check all axes and confirm it works on the hardware
-                    # if all(current_pos[a] != op for a, op in orig_pos.items()):
-                    #     axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
-                    #                                   axes=set(orig_pos.keys()), atol_linear=1e-6,
-                    #                                   atol_rotation=numpy.radians(6))
-                    #     if axes_updated:
-                    #         logging.debug("Position has updated fully: from %s -> %s", orig_pos,
-                    #                       current_pos)
-                    #         break
-
-                    # TODO Remove the check only on x any y
                     # Note that we don't use almost_equal() on the floats, because all we care about is whether
-                    # the position has changed. Even a very tiny change is a sign we've receive a position update.
+                    # the position has changed. Even a very tiny change is a sign we've received a position update.
                     if all(current_pos[a] != op for a, op in orig_pos.items()):
-                        # For x, y, and z, be extra picky and wait until they are "almost" at the target
-                        # Due to y-z linkage, there is an equivalent change in z axis when y axis is changed
-                        # Therefore, it is not important to check z
-                        # check for x and y axes as they move repeatedly in overview acquisitions
-                        axes_to_check = {"x", "y"}.intersection(target_pos.keys())
-                        if axes_to_check:
-                            axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
-                                                          axes=axes_to_check, atol_linear=1e-6)
+                        if linear_axes_to_check:
+                            linear_axes_updated = isNearPosition(current_pos=current_pos, target_position=target_pos,
+                                                                 axes=linear_axes_to_check, atol_linear=tol_linear)
+                        else:
+                            linear_axes_updated = True  # No axis to check => move is done
+
+                        if rotational_axes_to_check:
+                            rotational_axes_updated = isNearPosition(current_pos=current_pos,
```

Comment:
This doesn't really work because isNearPosition considers that an axis is a rotational axis if its name starts with a "r". So "t" will be considered linear. I think the simplest is to rename the "t" to "rx" before calling `isNearPosition`. Or you could add an extra parameter to `isNearPosition` to indicate the list of axes which are rotational.